### PR TITLE
Add CUTEst to the consistency tests and fix other models

### LINKED ISSUE #5: Test all functions on all interfaces
- Models
  - [x] SimpleNLPModel
  - [x] JuMPModel
  - [x] AMPLModel
  - [x] CUTEstModel
 - Functions (and inplace versions)
  - [x] obj(nlp, x)
  - [x] grad(nlp, x)
  - [x] hess(nlp, x)
  - [x] hess(nlp, x, y=y)
  - [x] hprod(nlp, x, v)
  - [x] hprod(nlp, x, v, y=y)
  - [x] cons(nlp, x)
  - [x] jac(nlp, x)
  - [x] jprod(nlp, x)
  - [x] jtprod(nlp, x)

### PR DESCRIPTION
**Pull request by abelsiqueira**
Change the size of genrose to 500 to match default GENROSE.SIF.
Fix HS11 and HS14 to match the Hock-Schittkowski paper.
Remove GENROSE because it is too slow.
Add BROWNDEN.
Add git depth following

    https://github.com/travis-ci/travis-ci/issues/4942

Closes #5

### CODE REVIEW

File: test/genrose.jl [hunk 1]
```diff
@@ -54,12 +54,20 @@ function genrose(n :: Int=100)
...
+    for i = 1:n-1
+      s += 100 * (x[i+1]-x[i]^2)^2 + (x[i]-1)^2
+    end
+    return s
+  end
 
   return SimpleNLPModel(x0, f)
 end
+
+cutest_problem_name = "GENROSE"
```
Comment (dpo) [inline test/genrose.jl]: This is already available from the function name (we just have to make it uppercase).

Comment (abelsiqueira) [inline test/genrose.jl]: This is here because of the other problems.

Comment (dpo) [inline test/genrose.jl]: Sorry I'm not sure I understand. What I'm saying is that the function name is available as a symbol, so we can just convert it uppercase if that's really necessary.

Comment (abelsiqueira) [inline test/genrose.jl]: I meant because of HS. hs005 is HS5 os CUTEst. Should I change the names?

Comment (dpo) [inline test/genrose.jl]: I see. Let's make the names all the same to simplify testing.

**Comment by abelsiqueira** [comment]
Updated

**Pull request merged**